public_fips_dropin_environments/python3_sklearn/Dockerfile

```diff
@@ -27,28 +27,29 @@ COPY --from=build /usr/bin/pip /usr/bin/pip
 # Just for convenience
 COPY --from=build /bin/ls /bin/ls
 
-COPY --chown=nonroot:nonroot requirements.txt requirements.txt
+COPY requirements.txt requirements.txt
 
-USER nonroot
+ENV VIRTUAL_ENV=/opt/venv
 
-RUN python -m ensurepip --default-pip && \
-    python -m pip install --upgrade --user pip && \
-    python -m pip install --user --no-cache-dir -r requirements.txt
+RUN sh -c "python -m venv ${VIRTUAL_ENV} && \
+    . ${VIRTUAL_ENV}/bin/activate && \
+    python -m ensurepip --default-pip && \
+    python -m pip install --upgrade pip && \
+    python -m pip install --no-cache-dir -r requirements.txt"
 
-# MARK: FUNCTIONAL-TEST-ADD-HERE. (This line is used by DRUM functional test automation and can be safely ignored.)
-
-ENV HOME=/home/nonroot
-ENV PATH=${HOME}/.local/bin:${PATH}
+ENV PATH=${VIRTUAL_ENV}/bin:${PATH}
+ENV HOME=/opt
```

Comment:
Surprised that having $HOME as /opt does no cause issues (we use this even now in production)

Overall I would like us to make sure that in a custom model you can write to /opt/code from the custom.py (to verify needed permissions).

Reply:
It is supposed to work, because the backend changes ownership recursively. But as you said - we better test it.

Reply:
Yeah, I've run up against the functional tests in this repo vs the way _things actually work in DR Platform_ related to this. Basically the `Dockerfile` (and thus image) you build from this repo isn't exactly what will be built and run in DataRobot because in DR, we layer on more commands that chmod/chown recursively the contents of `/opt/code` and also in DR we always run as user `1000:1000` where in tests in this repo, I think we run as the current UID of the user running the tests?

Reply:
Yeah right we just override all the relevant paths to the user we want in DR( 1000:1000 )